src/unix.rs

```diff
@@ -368,6 +395,84 @@ impl Client {
     }
 }
 
+// This should be available for all linux targets,
+// though only [`non_blocking_read`] currently uses it so adding gnu cfg.
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod linux {
+    use super::*;
+
+    use libc::{iovec, off_t, ssize_t, syscall, SYS_preadv2};
+
+    const RWF_NOWAIT: c_int = 0x00000008;
```

Comment:
That seems a little iffy. Theoretically the values could vary between architectures.
Musl added that flag recently. Can you leave a not to use that in the future once libc has it?

Reply:
Thanks, added a TODO for this.